compiler/rustc_typeck/src/astconv/mod.rs

```diff
@@ -2772,6 +2772,44 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
         Some(r)
     }
 
+    /// make sure that we are in the condition to suggest the blanket implementation, if we are in the
+    /// case of suggest it, use the function `err_reporter` to report the error or suggestion.
+    // FIXME: The copy of TyCtxt it is a bad think in this case?
+    fn maybe_lint_blanket_trait_impl(
+        &self,
+        self_ty: &hir::Ty<'_>,
+        err_reporter: fn(TyCtxt<'tcx>, &hir::Ty<'_>, &str, &str, Vec<(Span, String)>),
+    ) {
```

Comment:
Instead of passing in a reporter function, can this function return `Option<(&str, &str, Vec<Span, String>)>`?

Alternatively, you could pass in a `&mut DiagnosticBuilder<'_, _>` from the caller, and call `multipart_suggestion` on that builder.

Reply:
I will try to go with the `DiagnosticBuilder` solution!